### PR TITLE
feature-cmake-docs: baremetal clean-up

### PR DESCRIPTION
**Pull request by 0xc0170**
This is based on #1428

Only 2 commits are relevant here:
baremetal: move python tools related info to build-system b2500d6
build tools: move small lib selection to build-system 2c266c1

We need to add CMake baremetal/c lib via separate PR.

### CODE REVIEW

**Comment by 0xc0170** [comment]
> We need to add CMake baremetal/c lib via separate PR.

I'll fix it now here as it's under "baremetal" section.

**Comment by 0xc0170** [comment]
@rajkan01 Please review. Rebased

File: docs/bare_metal/using_bare_metal.md [hunk 1]
```diff
@@ -11,99 +11,9 @@ Here is a code snippet that can work for both Mbed OS profiles; it prints text a
 
 <span class="notes">**Note:** To be compatible with Arm microlib, a bare metal application should not return from `main()`. In this example, the `queue.dispatch_forever()` call never returns. For more details, see [Non-returning main()](../bare-metal/using-small-c-libraries.html).</span>
 
-## 1. Creating a bare metal application
+## How to switch to baremetal
```
Comment (0xc0170) [inline docs/bare_metal/using_bare_metal.md]: How to enable baremetal or rather use ?

**Review by rajkan01 — APPROVED** [review]
LGTM

File: docs/bare_metal/using_bare_metal.md (hunk 1 above)
Comment (rajkan01) [reply docs/bare_metal/using_bare_metal.md]: `Creating an application with a baremetal profile` as the actual baremetal.md describe the same

Comment (0xc0170) [reply docs/bare_metal/using_bare_metal.md]: I reverted it to the previous name

**Pull request merged**